### ISSUE TITLE
Require authentication on remote clients

### ISSUE THREAD

**Issue by KSXGitHub** [FredrikNoren/ungit#988]
Let's say that my computer is connected to a local area network (such as wifi), my own ip address is `192.168.1.5` and `::ffff:192.168.1.5`. I run `ungit --port=8448` so every device in my LAN can go to `http://192.168.1.5:8448` and edit my repo. I could just use `--allowedIPs` but it's more convenient to be able to access my repo from other devices.

### Request

1. The following hosts should be able to freely access ungit without authentication:
  * `127.0.0.1`
  * `::1`
  * `::ffff:127.0.0.1`
  * `192.168.1.5`
  * `::ffff:192.168.1.5`

2. The other hosts must login in order to use ungit in my machine

**Comment by gigi206** [comment]
I agree, for security reasons ungit must only allow local adresses.

How exactly works these options **--allowedIPs** and **--authentication --users** ?

I tried :
```
--allowedIPs='localhost'
--allowedIPs="['localhost']"
```

But I obtain the following message :
_This host is not authorized to connect
You are trying to connect to an Ungit instance from an unathorized host._

**Comment by jung-kim** [comment]
`--allowedIPs="['localhost']"` should be the correct syntax.  However, the reason why that isn't working is because there is because ungit doesn't resolve given [`allowedIPs`](https://github.com/FredrikNoren/ungit/blob/master/source/server.js#L88-L98).  

yes this is very rudimentary and not pedantic at all and we could introduce the [ip](https://www.npmjs.com/package/ip) npm module to do more precise matching but frankly it was a feature that wasn't used quite often so it was bit neglected but will fix it up soon if no body touches it by the time I get to it.

**Comment by gigi206** [comment]
For my point of view the main problem is a security issue : Ungit must bind by default on 127.0.0.1 and not 0.0.0.0

**Comment by jung-kim** [comment]
Oh wow, I thought it would default bind to 127.0.0.1 but I [guess not](https://nodejs.org/api/net.html#net_server_listen_port_host_backlog_callback).  Has this changed or has it always been like this for the `http`?  

I think there is a better way to start the server with express now but will visit that later.

@gigi206 you are right this did surprise me and I've totally missed your point.  Is there anything else I could add?

**Comment by gigi206** [comment]
No juste bind by default on 127.0.0.1 is good in my point of view :)

You can check with **lsof** on linux : 
```
$ lsof -i:8448
```
Actually you can see __TCP *:8448 (LISTEN)__ and it must be **TCP 127.0.0.1:8448 (LISTEN)**

Or **netstat** on Windows :
```
$ netstat -a
```

**Comment by jung-kim** [comment]
Yeah it's all localhost for me now so that's good.  Thanks!

**Comment by campersau** [comment]
Fixed in ungit 1.4.29 and made configurable in 1.4.30.
